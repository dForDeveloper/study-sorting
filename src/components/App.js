import React, { Component } from 'react';
import '../styles/main.scss';
import Demo from './Demo.js';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1 className="App--h1">study sorting</h1>
        <main>
          <Demo />
        </main>
      </div>
    );
  }
}

export default App;