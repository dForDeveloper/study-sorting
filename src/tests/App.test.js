import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot when showBubbleSort is true', () => {
    expect(wrapper.state()).toEqual({ showBubbleSort: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('it should have default state', () => {
    expect(wrapper.state()).toEqual({ showBubbleSort: true });
  });
  
  it('it should set showBubbleSort to false when right arrow is clicked',
  () => {
    expect(wrapper.state()).toEqual({ showBubbleSort: true });
    wrapper.find('.main--right-arrow').simulate('click');
    expect(wrapper.state()).toEqual({ showBubbleSort: false });
  });
  
  it('should match the snapshot when showBubbleSort is false', () => {
    expect(wrapper.state()).toEqual({ showBubbleSort: true });
    wrapper.find('.main--right-arrow').simulate('click');
    expect(wrapper.state()).toEqual({ showBubbleSort: false });
    expect(wrapper).toMatchSnapshot();
  });

  it('it should set showBubbleSort to true when left arrow is clicked',
  () => {
    expect(wrapper.state()).toEqual({ showBubbleSort: true });
    wrapper.find('.main--right-arrow').simulate('click');
    expect(wrapper.state()).toEqual({ showBubbleSort: false });
    wrapper.find('.main--left-arrow').simulate('click');
    expect(wrapper.state()).toEqual({ showBubbleSort: true });
  });
});