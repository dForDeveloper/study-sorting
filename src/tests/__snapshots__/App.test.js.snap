// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot when showBubbleSort is false 1`] = `
<div
  className="App"
>
  <h1
    className="App--h1 fade-in"
  >
    study sorting
  </h1>
  <main
    className="main"
  >
    <div
      className="main--left-arrow"
      onClick={[Function]}
    />
    <Demo
      algorithmName="Insertion Sort"
    />
  </main>
</div>
`;

exports[`App should match the snapshot when showBubbleSort is true 1`] = `
<div
  className="App"
>
  <h1
    className="App--h1 fade-in"
  >
    study sorting
  </h1>
  <main
    className="main"
  >
    <Demo
      algorithmName="Bubble Sort"
    />
    <div
      className="main--right-arrow"
      onClick={[Function]}
    />
  </main>
</div>
`;
